docs/proposals/formatted-output.md

```diff
@@ -0,0 +1,375 @@
+
+```console
+$ oras manifest fetch $REGISTRY/$REPO:$TAG --output sample-manifest --format go-template='{{.content.config.mediaType}}'
+application/vnd.oci.empty.v1+json
+```
+
+View the content of the generated manifest within specified `sample-manifest` file. The output should be compact JSON data:
+
+```console
+$ cat sample-manifest
+{"schemaVersion":2,"mediaType":"application/vnd.oci.image.manifest.v1+json","artifactType":"application/vnd.unknown.artifact.v1","config":{"mediaType":"application/vnd.oci.empty.v1+json","digest":"sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a","size":2,"data":"e30="},"layers":[{"mediaType":"application/vnd.oci.image.layer.v1.tar","digest":"sha256:6cb759c4296e67e35b0367f3c0f51dfdb776a0c99a45f39d0476e43d82696d65","size":14477,"annotations":{"org.opencontainers.image.title":"sbom.spdx"}},{"mediaType":"application/vnd.oci.image.layer.v1.tar","digest":"sha256:54c0e84503c8790e03afe34bfc05a5ce45c933430cfd9c5f8a99d2c89f1f1b69","size":6639,"annotations":{"org.opencontainers.image.title":"scan-test-verify-image.json"}}],"annotations":{"org.opencontainers.image.created":"2023-12-15T09:41:54Z"}}
+```
+
+> [!NOTE]
+> - `--output -` and `--format` can not be used at the same time due to conflicts.
+
+- Example: use `--format json` to print the metadata output in prettified JSON:
+
+```bash
+oras manifest fetch $REGISTRY/$REPO:$TAG --format json
+```
+
+```json
+{
+  "reference": " $REGISTRY/$REPO@sha256:8be4c36a29979c72fdd225654498791fb381a7dd8332ade1981274a16220fe1c",
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "digest": "sha256:8be4c36a29979c72fdd225654498791fb381a7dd8332ade1981274a16220fe1c",
+  "artifactType": "application/vnd.unknown.artifact.v1",
+  "content": {
+    "config": {
+      "mediaType": "application/vnd.oci.empty.v1+json",
+      "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
+      "size": 2,
+    },
+    "layers": [
+      {
+        "mediaType": "application/vnd.oci.image.layer.v1.tar",
+        "digest": "sha256:6cb759c4296e67e35b0367f3c0f51dfdb776a0c99a45f39d0476e43d82696d65",
+        "size": 14477,
+        "annotations": {
+          "org.opencontainers.image.title": "sbom.spdx"
+        }
+      },
+      {
+        "mediaType": "application/vnd.oci.image.layer.v1.tar",
+        "digest": "sha256:54c0e84503c8790e03afe34bfc05a5ce45c933430cfd9c5f8a99d2c89f1f1b69",
+        "size": 6639,
+        "annotations": {
+          "org.opencontainers.image.title": "scan-test-verify-image.json"
+        }
+      }
+    ],
+  "annotations": {
+    "org.opencontainers.image.created": "2023-12-15T09:41:54Z"
+  }
+},
+}
+```
+
+- Example: use `--template '{{ GO_TEMPLATE }}'` to manipulate the output data using Go template based on the chosen JSON data format `--format json`.
```

Comment:
Do you mean `--format json --template JSME_PATH` or `--format json --template JSON_PATH`?

Also we should promote the usage of `--format go-template=GO_TEMPLATE` as primary in our docs like `kubectl`

Reply:
I removed this example since this could be a new enhancement in the next iteration as discussed in https://github.com/oras-project/oras/pull/1199#discussion_r1579774298